streams/src/main/java/org/apache/kafka/streams/state/Stores.java

```diff
@@ -105,19 +106,28 @@ public StateStoreSupplier build() {
                                     private int numSegments = 0;
                                     private long retentionPeriod = 0L;
                                     private boolean retainDuplicates = false;
+                                    private boolean sessionWindows;
                                     private boolean logged = true;
 
                                     @Override
-                                    public PersistentKeyValueFactory<K, V> windowed(final long windowSize, long retentionPeriod, int numSegments, boolean retainDuplicates) {
+                                    public PersistentKeyValueFactory<K, V> windowed(final long windowSize, final long retentionPeriod, final int numSegments, final boolean retainDuplicates) {
                                         this.windowSize = windowSize;
                                         this.numSegments = numSegments;
                                         this.retentionPeriod = retentionPeriod;
                                         this.retainDuplicates = retainDuplicates;
+                                        this.sessionWindows = false;
 
                                         return this;
                                     }
 
                                     @Override
+                                    public PersistentKeyValueFactory<K, V> sessionWindows(final long retentionPeriod) {
```

Comment:
i think `sessionWindowed` is ok, but renaming `windowed` to `timeWindowed` is a breaking API change. Can we do that without a KIP?